### ISSUE TITLE
no aspect is given in infer_example

### ISSUE THREAD

**Issue by asarvi** [songyouwei/ABSA-PyTorch#79]

**Comment by asarvi** [comment]
hello, this aspect sentiment analyzer project is really great!the infer_example file is a big help. but i do have a question which i can not find its answer! why the "aspect" of input is "null" ? 
i changed the code a little bit so i can give the aspect manually to the 'evaluate' method as below :
`def evaluate(self, raw_texts , aspect ):                                                  
    context_seqs = [self.tokenizer.text_to_sequence(raw_texts.lower().strip())]           
    aspect_seqs = [self.tokenizer.text_to_sequence(aspect.lower().strip())]               
                                                                                          
    context_indices = torch.tensor(context_seqs, dtype=torch.int64).to(self.opt.device)   
    aspect_indices = torch.tensor(aspect_seqs, dtype=torch.int64).to(self.opt.device)     
    t_inputs = [context_indices, aspect_indices]                                          
    t_outputs = self.model(t_inputs)                                                      
                                                                                          
    t_probs = F.softmax(t_outputs, dim=-1).cpu().numpy()                                  
    return t_probs                                                                        
                                                                                          `

 but it looks like it has no effect on the output! for example i run:

`inf = Inferer(opt)                                                            
t_probs = inf.evaluate(' the charger is great but design is awful', 'design') `
the output is [1] but it should be -1 because the aspect that i want to know its polarity is negative!i tested this with many examples it looks the answer has nothing to do with the given aspect!
can you please explain what is the problem?

**Comment by GeneZC** [comment]
What model have you used?

**Comment by asarvi** [comment]
@GeneZC I tired memnet , ian and also atae_lstm! Same result for all appeared

**Comment by GeneZC** [comment]
Then try TNet instead.

**Comment by asarvi** [comment]
I did . in this model the third input parameter is aspect_in_text. I gave the aspect as input but this error appeared:
'str' object has no attribute 'cpu'
what should be the type of aspect_in_text?
thanks for your patience!

**Comment by GeneZC** [comment]
https://github.com/songyouwei/ABSA-PyTorch/blob/099e62b3f63a232bb4338928686a7abf8915cdea/data_utils.py#L146
Please refer to this line and the script for better understanding.

**Comment by asarvi** [comment]
thank you for your response I'm working with aoa model and it looks fine now
here is my evaluate function in infer_example
 ```
  def evaluate(self, raw_texts, aspect):
        text_left, _, text_right = raw_texts.lower().strip().partition("$t$")
        print(text_right)

        text_raw_indices = [self.tokenizer.text_to_sequence(text_left + " " + aspect + " " + text_right)]
        aspect_indices = [self.tokenizer.text_to_sequence(aspect)]
        context_indices = torch.tensor(text_raw_indices, dtype=torch.int64).to(self.opt.device)
        aspect_indices = torch.tensor(aspect_indices, dtype=torch.int64).to(self.opt.device)
        t_inputs = [context_indices, aspect_indices]
        t_outputs = self.model(t_inputs)

        t_probs = F.softmax(t_outputs, dim=-1).cpu().numpy()
        return t_probs
```
